cwms-data-api/src/main/java/cwms/cda/data/dao/location/kind/OutletDao.java

```diff
@@ -0,0 +1,240 @@
+import usace.cwms.db.jooq.codegen.udt.records.STR_TAB_T;
+import usace.cwms.db.jooq.codegen.udt.records.STR_TAB_TAB_T;
+
+public class OutletDao extends JooqDao<Outlet> {
+    private static final FluentLogger LOGGER = FluentLogger.forEnclosingClass();
+    private static final String RATING_LOC_GROUP_CATEGORY = "Rating";
+
+    public OutletDao(DSLContext dsl) {
+        super(dsl);
+    }
+
+    public List<Outlet> retrieveOutletsForProject(String projectId, String officeId) {
+        return connectionResult(dsl, conn -> {
+            Configuration config = DSL.using(conn).configuration();
+            Map<String, List<CompoundOutletRecord>> compoundOutlets = retrieveCompoundOutletsForProject(config,
+                                                                                                        projectId,
+                                                                                                        officeId);
+            LOCATION_REF_T locRef = LocationUtil.getLocationRef(projectId, officeId);
+
+            LocationGroupDao locGroupDao = new LocationGroupDao(dsl);
+            List<LocationGroup> groups = locGroupDao.getLocationGroups(officeId, RATING_LOC_GROUP_CATEGORY);
+
+            return CWMS_OUTLET_PACKAGE.call_RETRIEVE_OUTLETS(config, locRef)
+                                      .stream()
+                                      .map(struct -> mapToOutlet(struct, compoundOutlets, groups))
+                                      .collect(Collectors.toList());
+        });
+    }
+
+    public Outlet retrieveOutlet(String locationId, String officeId) {
+        return connectionResult(dsl, conn -> {
+            LOCATION_REF_T locRef = LocationUtil.getLocationRef(locationId, officeId);
+            Configuration config = DSL.using(conn).configuration();
+            PROJECT_STRUCTURE_OBJ_T outletStruct = CWMS_OUTLET_PACKAGE.call_RETRIEVE_OUTLET(config, locRef);
+
+            CwmsId projectId = LocationUtil.getLocationIdentifier(outletStruct.getPROJECT_LOCATION_REF());
+
+            Map<String, List<CompoundOutletRecord>> records = new HashMap<>();
+            try {
+                List<CompoundOutletRecord> compoundOutletRecords = retrieveCompoundOutlet(config, locationId,
+                                                                                          projectId.getName(),
+                                                                                          officeId);
+                records.put(locationId, compoundOutletRecords);
+            } catch (DataAccessException e) {
+                if (isNotFound(e)) {
+                    LOGGER.atFinest().withCause(e).log("No compound outlet records for outlet " + locationId);
+                } else {
+                    throw e;
+                }
+            }
+
+
+            LocationGroupDao locGroupDao = new LocationGroupDao(dsl);
+            List<LocationGroup> groups = locGroupDao.getLocationGroups(officeId, RATING_LOC_GROUP_CATEGORY);
```

Comment:
A third DB call made here

Reply:
This will break down to 2 retrievals once I make my virtual outlet changes.
retrieveOutletsForProject does 1 retrieval for location groups before retrieving the outlets for the project.